src/utils/config.ts

```diff
@@ -6,20 +6,42 @@
  * Does not mutate original object
  */
 export async function injectEnv<C extends string | Record<PropertyKey, any>>(config: C, notFoundValue: any = "") {
-	// Use simple regex replace for now, will see if object traversal and recursion is needed here (e.g: for non-serializable objects)
+	return injectVariables(config, { env: process.env }, notFoundValue)
+}
 
+/**
+ * Deeply injects variables into a configuration object/string/json
+ *
+ * Uses VSCode's variables reference pattern: https://code.visualstudio.com/docs/reference/variables-reference#_environment-variables
+ *
+ * Does not mutate original object
+ *
+ * There is a special handling for a nested (record-type) variables, where it is replaced by `propNotFoundValue` (if available) if the root key exists but the nested key does not.
+ *
+ * Matched keys that have `null` | `undefined` values are treated as not found.
+ */
+export async function injectVariables<C extends string | Record<PropertyKey, any>>(
+	config: string | Record<PropertyKey, any>,
```

Comment:
The type signature allows `Record<PropertyKey, any>` but the implementation uses `Object.entries()` which assumes string keys. Consider making the typing more precise:

```suggestion
export async function injectVariables<C extends string | Record<string, any>>(
	config: string | Record<string, any>,
```

This would better reflect the actual usage and prevent potential runtime issues.